libs/wingsdk/src/target-tf-aws/website.ts

```diff
@@ -126,7 +126,9 @@ export class Website extends cloud.Website {
       policy: allowDistributionReadOnly.json,
     });
 
-    this._url = distribution.domainName;
+    const url = distribution.domainName;
+
+    this._url = url.match(/^(.*):\/\/(.*)/) ? url : `https://${url}`;
```

Comment:
Do we really need this test? I think domain name by definition doesn't have a protocol, no?

Reply:
I think you're right, I just wonder what will happen when we'll let the user apply their own domain, instead of having the default cloudFront one... I think it's ok to be on the safe side here

Reply:
Then the check should be when we process user input. It's okay to assume that `domainName` is just a domain name and not a URL.

As a rule of thumb, we want to maintain the integrity of an object on the surface area.

For example:

```js
class MyClass {
  positiveNumber: num;

  setNumber(x: num) {
    // verify that the "invariant" is kept and `positiveNumber` is always positive.
    if (x < 0) { throw("error"); }

    this.positiveNumber = x;
  }

}
```

Reply:
ok, removed the test